app/src/components/tables/DeploysTable/DeploysTable.test.tsx

```diff
@@ -0,0 +1,134 @@
+  });
+
+  it('should render Total Rows', () => {
+    const totalRows = screen.getByTestId('total-rows');
+    expect(totalRows).toHaveTextContent('1 total rows');
+  });
+
+  it('should render a truncated Deploy Hash', () => {
+    const deployHash = screen.getByTestId('deploy-hash-link');
+    expect(deployHash).toHaveTextContent(
+      truncateHash(processedSidecarDeploys[0].deployHash),
+    );
+  });
+
+  it('should render a truncated Block Hash', () => {
+    const blockHash = screen.getByTestId('block-hash-link');
+    expect(blockHash).toHaveTextContent(
+      truncateHash(processedSidecarDeploys[0].blockHash),
+    );
+  });
+
+  it('should render a truncated Public Key', () => {
+    const publicKey = screen.getByTestId('public-key-link');
+    expect(publicKey).toHaveTextContent(
+      truncateHash(processedSidecarDeploys[0].publicKey),
+    );
+  });
+
+  it('should render an Age', () => {
+    const timestamp = screen.getByTestId('timestamp');
+    expect(timestamp).toHaveTextContent(
+      formatTimeAgo(new Date(processedSidecarDeploys[0].timestamp)),
+    );
+  });
+
+  it('should render a Contract type', () => {
+    const contractType = screen.getByTestId('contract-type');
+    expect(contractType).toHaveTextContent(
+      processedSidecarDeploys[0].contractType,
+    );
+  });
+
+  it('should render an Amount in motes and usd', () => {
+    const amountMotes = screen.getByTestId('amount-motes');
+    const amountUsd = screen.getByTestId('amount-usd');
+
+    expect(amountMotes).toHaveTextContent(
+      standardizeNumber(
+        (+processedSidecarDeploys[0].amount.motes / 10 ** 9).toFixed(3),
+      ).toString(),
+    );
+    expect(amountUsd).toHaveTextContent(
+      standardizeNumber(
+        (+processedSidecarDeploys[0].amount.usd / 10 ** 9).toFixed(3),
+      ).toString(),
+    );
+  });
+
+  it('should render Cost in motes and usd', () => {
+    const costMotes = screen.getByTestId('cost-motes');
+    const costUsd = screen.getByTestId('cost-usd');
+
+    expect(costMotes).toHaveTextContent(
+      standardizeNumber(
+        (+processedSidecarDeploys[0].cost.motes / 10 ** 9).toFixed(3),
+      ).toString(),
+    );
+    expect(costUsd).toHaveTextContent(
+      standardizeNumber(
+        (+processedSidecarDeploys[0].cost.usd / 10 ** 9).toFixed(3),
+      ).toString(),
+    );
+  });
+
+  it('should render a loading DeploysTable', () => {
+    render(
+      <Table
+        header={mockDeploysTableHeader}
+        columns={mockDeploysTableColumns}
+        data={processedSidecarDeploys}
+        footer={mockDeploysTableFooter}
+        tableBodyLoading
+        currentPageSize={mockDeploysTableOptions.pagination.pageSize}
+        placeholderData={{}}
+        isLastPage={totalPages === mockDeploysTableOptions.pagination.pageSize}
+      />,
+    );
+    const skeletonLoader = screen.getAllByTestId('skeleton-loader');
+    expect(skeletonLoader).toHaveLength(7);
```

Comment:
This is just the number of columns right? Based on my above comment maybe we should do something similar. I guess we could dynamically get this number like

```
const TOTAL_COLUMNS = 7; // assign above this block of code somewhere
expect(skeletonLoader).toHaveLength(totalRows.length * TOTAL_COLUMNS);
```

Does that make sense?

Reply:
Updated!